Subject: [PATCH] build: make blkdiscard configurable

blkdiscard is only wired up as UL_BUILD_INIT(...,[check]), so with
--disable-all-programs there is no way to ask for it and
--enable-blkdiscard is an unrecognized option. Give it an AC_ARG_ENABLE
like the other tools. Still not present upstream in 2.42.2.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -1843,6 +1843,14 @@
 UL_REQUIRES_LINUX([fstrim])
 UL_REQUIRES_BUILD([fstrim], [libmount])
 AM_CONDITIONAL([BUILD_FSTRIM], [test "x$build_fstrim" = xyes])
+
+AC_ARG_ENABLE([blkdiscard],
+  AS_HELP_STRING([--disable-blkdiscard], [do not build blkdiscard(8)]),
+  [], [UL_DEFAULT_ENABLE([blkdiscard], [check])]
+)
+UL_BUILD_INIT([blkdiscard])
+UL_REQUIRES_LINUX([blkdiscard])
+AM_CONDITIONAL([BUILD_BLKDISCARD], [test "x$build_blkdiscard" = xyes])
 
 
 AC_ARG_ENABLE([swapon],
@@ -2235,10 +2243,6 @@
 UL_REQUIRES_LINUX([fsfreeze])
 AM_CONDITIONAL([BUILD_FSFREEZE], [test "x$build_fsfreeze" = xyes])
 
-UL_BUILD_INIT([blkdiscard], [check])
-UL_REQUIRES_LINUX([blkdiscard])
-AM_CONDITIONAL([BUILD_BLKDISCARD], [test "x$build_blkdiscard" = xyes])
-
 UL_BUILD_INIT([blkzone], [check])
 UL_REQUIRES_LINUX([blkzone])
 UL_REQUIRES_HAVE([blkzone], [linux_blkzoned_h], [linux/blkzoned.h header])
